### PR TITLE
[ADHOC] fix: add owner param to BoostCore constructor

### PR DESCRIPTION
**Pull request by jonathandiep**
### Description
Adds an owner param to the BoostCore constructor

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 19a7a3b4d0500fde4ff48bfcaf485a0716ee4f5b
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e71e4af5365c7b5383e6f2d4d3071950a6107f1f
</p>

File: packages/evm/script/solidity/Deploy.s.sol [hunk 1]
```diff
@@ -29,8 +29,9 @@ contract CoreDeployer is ScriptUtils {
     }
 
     function _deployCore(address registry) internal returns (address core) {
+        address owner = vm.addr(vm.envUint("BOOST_CORE_OWNER_ADDRESS"));
```
Comment (topocount) [inline packages/evm/script/solidity/Deploy.s.sol:32]: This should use envAddr to save the extra conversion https://book.getfoundry.sh/cheatcodes/env-address?highlight=EnvAdd#signature

**Review by topocount — APPROVED** [review]
Looks good! one nit

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f9b858c7827ba38326070328ae8f885ce0d5fcb4
</p>

**Pull request merged**